src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/PeriodicSyncStrategy.java

```diff
@@ -139,6 +139,8 @@ private void syncLoop() {
                     }
                 } catch (InterruptedException e) {
                     logger.atWarn(SYNC_EVENT_TYPE).log("Interrupted while waiting for sync requests");
+                    // Add the sync request back in the queue since we still need to process this later.
+                    syncQueue.offer(request);
```

Comment:
Is there still a chance that the thread is interrupted but no `InterruptedException` is thrown? We check the `Thread.currentThread().isInterrupted()`  in the while loop here, and in the do / while loop in the RealtimeSync.

Would it be better to set the request to null after processing (at line 119) and before picking up the next one. Then outside the loop, before the`finally` at line 172 you could check:
```
if (request != null) {
  syncQueue.offer(request);
}
```